core/src/main/java/com/aem/demo/core/servlets/LikeDislikeServlet.java

```diff
@@ -31,14 +30,14 @@ public class LikeDislikeServlet extends SlingAllMethodsServlet {
     LikeDislikeService service;
 
     protected void doPost(final SlingHttpServletRequest request, final SlingHttpServletResponse response) throws IOException {
-        boolean reaction = Boolean.parseBoolean(request.getParameter("reaction"));
+        String reaction = request.getParameter("reaction");
         String path = request.getParameter("path");
         JSONObject responseData = new JSONObject();
         try {
-            if (reaction) {
+            if (reaction.equals(LIKE)) {
```

Comment:
Can get NullPointerException if the parameter "reaction" is missing. Which is very probably in an HTTP request.
Don't be careless) Think of consequences every time.
`StringUtils.equals(reaction, LIKE)`